From 4f713f3a0e2eda9c785647e24cb4ea80a4fb4794 Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Fri, 2 Feb 2024 19:05:35 +0530
Subject: [PATCH] hostapd: Update auth_alg in sta_info structure for partner
 links

Incase of FT roaming which leads to failing of ping traffic which is
due to the partner links auth_alg is not updated with WLAN_AUTH_FT
for partner links.

Add auth_alg for all the partner links

Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
 src/ap/ieee802_11.c | 1 +
 1 file changed, 1 insertion(+)

--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -4784,6 +4784,7 @@ static int ieee80211_ml_process_link(str
 
 	sta->flags |= WLAN_STA_ASSOC_REQ_OK;
 	sta->vlan_id = origin_sta->vlan_id;
+	sta->auth_alg = origin_sta->auth_alg;
 
 	/* TODO: What other processing is required? */
 
